src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Internal/PrecompileRunCommand.cs

```diff
@@ -236,10 +237,20 @@ private ViewCompilationInfo[] GenerateCode()
             Parallel.For(0, results.Length, ParalellOptions, i =>
             {
                 var fileInfo = files[i];
+                var templateEngine = MvcServiceProvider.TemplateEngine;
                 using (var fileStream = fileInfo.CreateReadStream())
                 {
-                    var csharpDocument = MvcServiceProvider.TemplateEngine.GenerateCode(fileInfo.ViewEnginePath);
-                    results[i] = new ViewCompilationInfo(fileInfo, csharpDocument);
+                    var csharpDocument = templateEngine.GenerateCode(fileInfo.ViewEnginePath);
+                    var compilationInfo = new ViewCompilationInfo(fileInfo, csharpDocument);
+                    if (PageDirectiveFeature.TryGetPageDirective(fileInfo.CreateReadStream, out var template))
```

Comment:
This is difficult to do. We pull the engine out of the design time service provider, so we can't configure it. Maybe we could stash it in the code document as part of matching: https://github.com/aspnet/Razor/blob/e2267f54a6db5da032d00730f4c1e150202fce1c/src/Microsoft.AspNetCore.Mvc.Razor.Extensions/RazorPageDocumentClassifierPass.cs#L18?